crates/primitives/Cargo.toml

```diff
@@ -14,9 +14,11 @@ reth-rlp = { path = "../common/rlp", features = [
     "derive",
     "ethereum-types",
 ] }
+reth-rlp-derive = { path = "../common/rlp-derive" }
 reth-codecs = { version = "0.1.0", path = "../storage/codecs" }
 
 # ethereum
+discv5 = { git = "https://github.com/sigp/discv5" }
```

Comment:
we don't want this here.

instead of moving the `kad_key` function to primitives, let's keep them in discv4 and convert it to a helper function `kad_key(&NodeRecord)` instead

Reply:
Okay. Just managed to remove the `discv5` dependency from _primitives_. Had to move `NodeKey` back to `discv4::node`, along with `kad_key` and `NodeRecord::key` (now a helper function, as suggested).